Description: Resolve issue with FTBFS on 32bit ppc platform
 .
 NoBarrier versions for Atomic64 can be implemented on PPC32
Origin: http://code.google.com/p/gperftools/issues/detail?id=431

Index: google-perftools/src/base/atomicops-internals-linuxppc.h
===================================================================
--- google-perftools.orig/src/base/atomicops-internals-linuxppc.h	2012-06-27 09:25:34.879532000 +0100
+++ google-perftools/src/base/atomicops-internals-linuxppc.h	2012-07-04 21:18:36.974406885 +0100
@@ -369,14 +369,20 @@
   return *ptr;
 }
 
-#ifdef __PPC64__
-
-// 64-bit Versions.
-
+// NoBarrier versions for Atomic64 can be implemented on PPC32
 inline void NoBarrier_Store(volatile Atomic64 *ptr, Atomic64 value) {
   *ptr = value;
 }
 
+inline Atomic64 NoBarrier_Load(volatile const Atomic64 *ptr) {
+  return *ptr;
+}
+
+
+#ifdef __PPC64__
+
+// 64-bit Versions.
+
 inline void Acquire_Store(volatile Atomic64 *ptr, Atomic64 value) {
   *ptr = value;
   // This can't be _lwsync(); we need to order the immediately
@@ -390,10 +396,6 @@
   *ptr = value;
 }
 
-inline Atomic64 NoBarrier_Load(volatile const Atomic64 *ptr) {
-  return *ptr;
-}
-
 inline Atomic64 Acquire_Load(volatile const Atomic64 *ptr) {
   Atomic64 value = *ptr;
   _lwsync();
